org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/eval/DebugEvaluationSetup.java

```diff
@@ -55,7 +60,10 @@ protected void setUp() {
 				LocalVMLauncher launcher;
 				try {
 					launcher = LocalVMLauncher.getLauncher();
-					launcher.setVMArguments(new String[]{"-verify"});
+					long vmVersion = Util.getMajorMinorVMVersion();
+					if (vmVersion < ClassFileConstants.JDK24) {
+						launcher.setVMArguments(new String[]{"-verify"});
```

Comment:
is option `-verify` being removed, or what kind of trouble does it cause?

Reply:
Not yet but will cause tons of tests to fail with this warning. We need a strategy if we continue to require this by our tests.

Reply:
> > Not yet but will cause tons of tests to fail with this warning. We need a strategy if we continue to require this by our tests.
> 
> Are you saying this from experience with previous versions, or is anything becoming stricter specifically in 24?
> 
> I'm just curious what could make classfiles generated by ecj fail the VM's verification?

This part, I don't know. I haven't spent time on this yet. I was hoping someone would already know the answer :) 
All I know now is the test are failing with unexpected warning about the -verify argument.

Reply:
>  All I know now is the test are failing with unexpected warning about the -verify argument.

Where could I such such failures?

Reply:
Unfortunately, the jenkins that failed with that error is gone. But here's what I did to reproduce locally - just set the "passed" to false after the call to `verifyClassFiles()` in line number 3574 of `AbstractRegressionTest`.

Reply:
> "Warning: -verify option is deprecated and may be removed in a future release."

Well ...

Actually, I can't find `-verify` in any documentation of the `java` command, recent or old.

[At 1.8 ](https://docs.oracle.com/javase/8/docs/technotes/tools/unix/java.html) I see "-Xverify:mode".

And even `-Xverify` is [deprecated since 13](https://www.oracle.com/java/technologies/javase/13all-relnotes.html)

It looks like verification will happen always, and hence the option is plain useless.

How to test: revert one of the recent fixes in JAVA_BETA23 that fixed a verify error, and run the test without -verify. It should still fail :)

Reply:
So what do you reckon? Just removing it altogether instead of doing it conditionally?